AK/Tuple.h

```diff
@@ -219,8 +243,34 @@ struct Tuple : Detail::Tuple<Ts...> {
 template<class... Args>
 Tuple(Args... args) -> Tuple<Args...>;
 
+template<size_t I, typename T>
+struct TupleElement;
+
+template<size_t I, typename... Ts>
+struct TupleElement<I, Tuple<Ts...>> {
+    using Type = TypeListElement<I, typename Tuple<Ts...>::Types>::Type;
+};
+
+template<typename T>
+struct TupleSize;
+
+template<typename... Ts>
+struct TupleSize<Tuple<Ts...>> : Detail::IntegralConstant<size_t, sizeof...(Ts)> { };
+
+}
+
+namespace std {
```

Comment:
Drive by comment, can't do a full review (on vacation)

But you can and should specialize these without entering the std namespace